lib/ansible/module_utils/hwc_utils.py

```diff
@@ -0,0 +1,317 @@
+# Copyright (c), Google Inc, 2017
+# Simplified BSD License (see licenses/simplified_bsd.txt or
+# https://opensource.org/licenses/BSD-2-Clause)
+
+try:
+    import requests
+    HAS_REQUESTS = True
+except ImportError:
+    HAS_REQUESTS = False
+
+try:
+    from keystoneauth1.adapter import Adapter
+    from keystoneauth1.identity import v3
+    from keystoneauth1 import session
+    HAS_THIRD_LIBRARIES = True
+except ImportError:
+    HAS_THIRD_LIBRARIES = False
+
+from ansible.module_utils.basic import AnsibleModule, env_fallback
+from ansible.module_utils._text import to_text
+
+
+def navigate_hash(source, path, default=None):
+    if not (source and path):
+        return None
+
+    key = path[0]
+    path = path[1:]
+    if key not in source:
+        return default
+    result = source[key]
+    if path:
+        return navigate_hash(result, path, default)
+    else:
+        return result
+
+
+class HwcRequestException(Exception):
+    pass
+
+
+def remove_empty_from_dict(obj):
+    return _DictClean(
+        obj,
+        lambda v: v is not None and v != {} and v != []
+    )()
+
+
+def remove_nones_from_dict(obj):
+    return _DictClean(obj, lambda v: v is not None)()
+
+
+# Handles the replacement of dicts with values -> the needed value for HWC API
+def replace_resource_dict(item, value):
+    if isinstance(item, list):
+        items = []
+        for i in item:
+            items.append(replace_resource_dict(i, value))
+        return items
+    else:
+        if not item:
+            return item
+        return item.get(value)
+
+
+def are_dicts_different(expect, actual):
+    # Remove all output-only from actual.
```

Comment:
You should use [docstrings](https://en.wikipedia.org/wiki/Docstring#Python) instead of comments to document functions (here and in other places).

Reply:
Done